torch/autograd/function.py

```diff
@@ -560,6 +585,13 @@ def bind_default_args(func, *args, **kwargs):
 
         return custom_function_call(cls, *args, **kwargs)
 
+    @staticmethod
+    def _compiled_autograd_key(ctx):
+        # TODO: figure out how to increment only once per compiled autograd graph
+        # otherwise we'll recompile every time for compiled autograd
+        from torch._functorch.aot_autograd import AOT_COUNTER
+        return (next(AOT_COUNTER),)
```

Comment:
This is wrong, it will return a new key every time it is called.  So you will *never* get a cache hit.